src/ui/qml/MessageDelegate.qml

```diff
@@ -158,18 +158,20 @@ Column {
                 visible: hoveredLink.length > 0
             }
             MenuItem {
-                text: qsTr("Copy")
-                shortcut: "Ctrl+C"
+                text: qsTr("Copy Message")
+                visible: textField.selectedText.length == 0 || textField.selectedText == hoveredLink
                 onTriggered: {
-                    if (textField.selectedText.length > 0) {
-                        textField.copy()
-                    } else {
-                        textField.selectAll()
-                        textField.copy()
-                        textField.deselect()
-                    }
+                    textField.selectAll()
+                    textField.copy()
+                    textField.deselect()
                 }
             }
+            MenuItem {
+                text: qsTr("Copy Selection")
+                visible: textField.selectedText.length > 0 && textField.selectedText != hoveredLink
```

Comment:
This comparison isn't always working (Qt 5.4, Mac). When mousing over the text `www.google.com`, hoveredLink is actually `http://www.google.com`.